server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -0,0 +1,898 @@
+                  currentCompactionId.set(eci);
+                  return coordinatorClient.getCompactionJob(TraceUtil.traceInfo(),
+                      getContext().rpcCreds(), queueName,
+                      ExternalCompactionUtil.getHostPortString(compactorAddress.getAddress()),
+                      eci.toString());
+                } catch (Exception e) {
+                  currentCompactionId.set(null);
+                  throw e;
+                } finally {
+                  ThriftUtil.returnClient(coordinatorClient);
+                }
+              }
+            });
+    return nextJobThriftCall.run();
+  }
+
+  /**
+   * Get the client to the CompactionCoordinator
+   *
+   * @return compaction coordinator client
+   * @throws TTransportException
+   *           when unable to get client
+   */
+  protected CompactionCoordinatorService.Client getCoordinatorClient() throws TTransportException {
+    HostAndPort coordinatorHost = ExternalCompactionUtil.findCompactionCoordinator(getContext());
+    if (null == coordinatorHost) {
+      throw new TTransportException("Unable to get CompactionCoordinator address from ZooKeeper");
+    }
+    LOG.trace("CompactionCoordinator address is: {}", coordinatorHost);
+    return ThriftUtil.getClient(COORDINATOR_CLIENT_FACTORY, coordinatorHost, getContext());
+  }
+
+  /**
+   * Create compaction runnable
+   *
+   * @param job
+   *          compaction job
+   * @param totalInputEntries
+   *          object to capture total entries
+   * @param totalInputBytes
+   *          object to capture input file size
+   * @param started
+   *          started latch
+   * @param stopped
+   *          stopped latch
+   * @param err
+   *          reference to error
+   * @return Runnable compaction job
+   */
+  protected Runnable createCompactionJob(final TExternalCompactionJob job,
+      final LongAdder totalInputEntries, final LongAdder totalInputBytes,
+      final CountDownLatch started, final CountDownLatch stopped,
+      final AtomicReference<Throwable> err) {
+
+    return new Runnable() {
+      @Override
+      public void run() {
+        // Its only expected that a single compaction runs at a time. Multiple compactions running
+        // at a time could cause odd behavior like out of order and unexpected thrift calls to the
+        // coordinator. This is a sanity check to ensure the expectation is met. Should this check
+        // ever fail, it means there is a bug elsewhere.
+        Preconditions.checkState(compactionRunning.compareAndSet(false, true));
+        try {
+          LOG.info("Starting up compaction runnable for job: {}", job);
+          updateCompactionState(job, TCompactionState.STARTED, "Compaction started");
+
+          final AccumuloConfiguration tConfig;
+
+          if (!job.getTableCompactionProperties().isEmpty()) {
+            tConfig = new ConfigurationCopy(DefaultConfiguration.getInstance());
```

Comment:
Looking how this works here and in the tserver code, thinking this is brittle and error prone. What if a non table props is used, like a general prop?  That could cause silent failures in this code.  Think we need to rework this, but not sure what to do ATM.  The peer code in the tserver is in CompactableImpl.reserveExternalCompaction()

Reply:
Resolved this in c2da878084c51d59acf6cbda9a92f59d629d6a2b